README.md

```diff
@@ -194,7 +203,24 @@ setTimestamp(Instant.now())
 
 - **flush**()
 
-Flushes the current MetricsContext to the configured sink and resets all properties, dimensions and metric values. The namespace and default dimensions will be preserved across flushes.
+Flushes the current MetricsContext to the configured sink and resets all properties and metric values. The namespace and default dimensions will be preserved across flushes. The default behavior is to preserve all custom dimensions as well, but this can be disabled by invoking `setFlushPreserveDimensions(false)`.
+
+Example:
+
+```java
+flush();  // default dimensions and custom dimensions will be preserved after each flush()
+```
+
+```java
+setFlushPreserveDimensions(false);
+flush();  // only default dimensions will be preserved after each flush()
+```
+
+```java
+setFlushPreserveDimensions(false);
+flush();
+resetDimensions(false);  // no dimensions will be preserved after each flush()
```

Comment:
By saying that "no dimensions will be preserved after each flush" makes it sound like you only need to invoke `resetDimensions()` once, instead of every time to not keep dimensions.